docs/pubsub.rst

```diff
@@ -468,12 +468,98 @@ Also you can disable built-in UDPv4 network transport using the attribute ``rtps
 
     part_attr.rtps.useBuiltinTransports = false;
 
-Network endpoints are defined by *eProsima Fast RTPS* as locators. Locators in *eProsima Fast RTPS* are enclosed as type :class:`Locator_t`, which has the following fields:
+Network endpoints are defined by *eProsima Fast RTPS* as locators.
+Locators in *eProsima Fast RTPS* are enclosed as type :class:`Locator_t`, which has the following fields:
 
-* ``kind``: Defines the protocol. *eProsima Fast RTPS* currently supports UDPv4 or UDPv6
+* ``kind``: Defines the protocol. *eProsima Fast RTPS* currently supports UDPv4, UDPv6, TCPv4 or TCPv6
 * ``port``: Port as an UDP/IP port.
 * ``address``: Maps to IP address
 
+If you are configuring your application using a XML file, you can add custom transports too, inside the profiles
+section.
+
+.. code-block:: xml
+
+    <profiles>
+        <transports>
+            <transport>
+                <transport_id>TCPv4_Server</transport_id>
+                <type>TCPv4</type>
+                <wan_addr>127.0.0.1</wan_addr>
+                <listening_ports>
+                    <port>5100</port>
+                </listening_ports>
+            </transport>
+        </transports>
+
+        <participant profile_name="TCPShapes">
+            <rtps>
+                <userTransports>
+                    <transport_id>TCPv4_Server</transport_id>
+                </userTransports>
+                ...
+            </rtps>
+        </participant>
+        ...
+    </profiles>
+
+To add a custom transport to a participant, you must indicate its transport_id inside the userTransports labels of the
+participant.
+
+You can learn more about profiles in xml in :ref:`xml-profiles`.
+
+TCP Transport Configuration
+"""""""""""""""""""""""""""
+
+To use TCP as transport you need to define some more configurations:
+
+First, you need to indicate Fast-RTPS to not use default transports (if you only want to use TCP) through
```

Comment:
"if you only want to use TCP" can create misunderstandings because this flag can be used with other transports or configurations

Reply:
You are right. I think the best option is to remove the full sentence, as *useBuiltinTransports* is explained already.